packages/common/src/intl/IntlProvider.tsx

```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import { IntlProvider as ReactIntlProvider } from 'react-intl';
+import {
+  LocaleMessages,
+  SupportedLocales,
+  importMessages,
+} from './intlHelpers';
+
+export const IntlProvider: React.FC<
+  Omit<React.ComponentProps<typeof ReactIntlProvider>, 'messages'> & {
+    messages?: LocaleMessages;
+  }
+> = props => {
+  const { locale } = props;
+  const [messages, setMessages] = React.useState<Record<string, string> | null>(
+    null
+  );
+  React.useEffect(() => {
+    importMessages(locale as SupportedLocales).then(setMessages);
+  }, [locale]);
+
+  return messages ? (
+    <ReactIntlProvider locale={locale} messages={messages} key={locale}>

```

Comment:
`key` is to encourage the component to re render - perhaps not necessary if we aren't using some kind of dynamic language selection